rust/tests/command_optimize.rs

```diff
@@ -307,7 +305,10 @@ async fn test_conflict_for_remove_actions() -> Result<(), Box<dyn Error>> {
     )
     .await?;
 
-    let maybe_metrics = plan.execute(dt.object_store(), &dt.state).await;
+    let maybe_metrics = plan
+        .execute(dt.object_store(), &dt.state, 1, 20, None)
+        .await;
```

Comment:
Could you add at least one test where we pass in a commit interval? even if it doesn't make an intermediate commit, it would be good to know those code paths run through.

Reply:
Added a test where an intermediate commit is actually created.